### PR TITLE
Remove `eth-keyring-controller` use from `MetamaskController`

### LINKED ISSUE #20510: Migrate all remaining `@metamask/eth-keyring-controller` interactions
We should migrate all the interactions left with `@metamask/eth-keyring-controller`:
- [x] #20722
- [x] #20723
- [x] #20724
- [x] #20725
- [ ] #20510
  - (MetamaskController)

### LINKED ISSUE #20512: Remove `@metamask/keyring-controller` patch

### PR DESCRIPTION
**Pull request by mikesposito**
## Explanation

This PR removes the leftover `EthKeyringController` interactions from `MetamaskController`.

- Updated `@metamask/signature-controller` to `^6.0.0` along with its patch
- Removed interactions with `@metamask/eth-keyring-controller` from `MetamaskController`
  - Now state passed to UI is taken from `@metamask/keyring-controller`
- Fixed behaviour for some tests, needed after this refactoring
  - `createNewVaultAndKeychain` test assertion slightly changed
  - `preferencesController.removeAddress` mocked to always resolve
- `MetamaskController.getState()` returns a flattened state object which now includes `coreKeyringController`'s state, but the `vault` property is removed

## References

* Fixes #20510
* Fixes #20512

## Manual Testing Steps

Interactions with keyrings should work as before

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [x] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by mikesposito** [comment]
@metamaskbot update-policies

File: app/scripts/metamask-controller.test.js [hunk 1]
```diff
@@ -730,6 +730,10 @@ describe('MetaMaskController', () => {
     describe('unlockHardwareWalletAccount', () => {
       const accountToUnlock = 10;
       beforeEach(async () => {
+        await metamaskController.coreKeyringController.createNewVaultAndRestore(
+          'password',
+          TEST_SEED,
+        );
```
Comment (mikesposito) [inline app/scripts/metamask-controller.test.js:736]: We can't really test anything properly without a vault and an unlocked `KeyringController`.

File: .yarn/patches/@metamask-keyring-controller-npm-8.0.0-806630ae4e.patch [hunk 2]
```diff
@@ -1,40 +1,23 @@
...
++    const { isUnlocked, keyrings, encryptionKey, encryptionSalt } = 
++        __classPrivateFieldGet(this, _KeyringController_keyring, "f").memStore.getState();
++
++    this.update(() => ({
++        vault,
++        keyrings,
++        isUnlocked,
++        encryptionKey,
++        encryptionSalt,
++    }));
```
Comment (mikesposito) [inline .yarn/patches/@metamask-keyring-controller-npm-8.0.0-806630ae4e.patch:20]: Without filtering out properties that come from `EthKeyringController`, `KeyringController` will throw an error for metadata not found for `keyringTypes`

This issue is solved upstream by [this PR](https://github.com/MetaMask/core/pull/1710)

File: package.json [hunk 3]
```diff
@@ -245,7 +245,7 @@
     "@metamask/design-tokens": "^1.12.0",
     "@metamask/desktop": "^0.3.0",
     "@metamask/eth-json-rpc-middleware": "^11.0.0",
-    "@metamask/eth-keyring-controller": "^10.0.1",
+    "@metamask/eth-keyring-controller": "^13.0.1",
```
Comment (mikesposito) [inline package.json:248]: This is kept here for `keyringBuilderFactory` only

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by mikesposito** [comment]
@SocketSecurity ignore @metamask/signature-controller@6.1.1

This is our package, `6.1.1` published ~1 hour ago

**Comment by mikesposito** [comment]
@metamaskbot update-policies

**Review by Gudahtt — APPROVED** [review]

**Review by legobeat — APPROVED** [review]

**Pull request merged**